src/js/components/ServiceDetailConfigurationTab.js

```diff
@@ -1,19 +1,100 @@
+import {Dropdown} from 'reactjs-components';
 import React from 'react';
 
 import ConfigurationView from './ConfigurationView';
+import DCOSStore from '../stores/DCOSStore';
 import Service from '../structs/Service';
 
+const METHODS_TO_BIND = [
+  'handleVersionSelection'
+];
+
 class ServiceDetailConfigurationTab extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      selectedVersionID: null
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    this.setState({
+      selectedVersionID: this.props.service.getVersion()
+    });
+    DCOSStore.fetchServiceVersions(this.props.service.getId());
+  }
+
+  handleVersionSelection(versionItem) {
+    this.setState({
+      selectedVersionID: versionItem.id
+    });
+  }
+
+  getVersionsDropdown() {
+    let {service} = this.props;
+    let versions = service.getVersions();
+
+    if (versions.size < 2) {
+      return null;
+    }
+
+    let versionItems = [];
+    for (let version of versions.keys()) {
```

Comment:
This would work, but is may a little to much... `Array.from(versions.keys()).map(function (key) { return key} );`